File: src/04.stack/01.682.ts
export function calPoints(operations: string[]): number {
    const stack = [];
    for (let i = 0; i < operations.length; i++) {
        let item = operations[i];

        if (item === "C") {
            stack.pop();
        } else if (item === "D") {
            stack.push(stack[stack.length - 1] * 2);
        } else if (item === "+") {
            stack.push(stack[stack.length - 1] + stack[stack.length - 2]);
        } else {
            stack.push(Number(item));
        }
    }

    let count = 0;

    for (let i = 0; i < stack.length; i++) {
        count += stack[i];
    }

    return count;
}

File: src/04.stack/01.682.go
package stack

import "strconv"

func CalPoints(operations []string) int {
	stack := []int{}

	for _, v := range operations {
		if v == "C" {
			stack = stack[:len(stack)-1]
		} else if v == "D" {
			stack = append(stack, stack[len(stack)-1]*2)
		} else if v == "+" {
			stack = append(stack, stack[len(stack)-1]+stack[len(stack)-2])
		} else {
			num, _ := strconv.Atoi(v)
			stack = append(stack, num)
		}
	}

	count := 0
	for _, v := range stack {
		count += v
	}

	return count
}
